components/brave_wallet/browser/simulation_service.cc

```diff
@@ -0,0 +1,326 @@
+                                       brave_wallet::mojom::CoinType coin) {
+  return (coin == brave_wallet::mojom::CoinType::SOL &&
+          chain_id == brave_wallet::mojom::kSolanaMainnet) ||
+         (coin == brave_wallet::mojom::CoinType::SOL &&
+          chain_id == brave_wallet::mojom::kSolanaTestnet) ||
+         (coin == brave_wallet::mojom::CoinType::SOL &&
+          chain_id == brave_wallet::mojom::kSolanaDevnet) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kMainnetChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kGoerliChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kPolygonMainnetChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kBinanceSmartChainMainnetChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kArbitrumMainnetChainId);
+}
+
+bool HasMessageScanSupportInternal(const std::string& chain_id,
+                                   brave_wallet::mojom::CoinType coin) {
+  return (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kMainnetChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kGoerliChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kPolygonMainnetChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kBinanceSmartChainMainnetChainId) ||
+         (coin == brave_wallet::mojom::CoinType::ETH &&
+          chain_id == brave_wallet::mojom::kArbitrumMainnetChainId);
+}
+
+// Function to convert all numbers in JSON string to strings.
+//
+// For sample JSON response, refer to ParseJupiterQuote.
+absl::optional<std::string> ConvertAllNumbersToString(const std::string& json) {
+  auto converted_json = std::string(json::convert_all_numbers_to_string(json));
+  if (converted_json.empty()) {
+    return absl::nullopt;
+  }
+
+  return converted_json;
+}
+
+base::flat_map<std::string, std::string> GetHeaders() {
+  return {
+      {brave_wallet::kBlowfishAPIVersionHeader,
+       brave_wallet::kBlowfishAPIVersion},
+      {net::HttpRequestHeaders::kContentType, "application/json"},
+      {brave_wallet::kBraveServicesKeyHeader, BUILDFLAG(BRAVE_SERVICES_KEY)}};
+}
+
+}  // namespace
+
+namespace brave_wallet {
+
+GURL SimulationService::base_url_for_test_;
+
+SimulationService::SimulationService(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    JsonRpcService* json_rpc_service)
+    : api_request_helper_(GetNetworkTrafficAnnotationTag(), url_loader_factory),
+      json_rpc_service_(json_rpc_service),
+      weak_ptr_factory_(this) {
```

Comment:
prefer to initialize in header instead: 
`base::WeakPtrFactory<SimulationService> weak_ptr_factory_{this};`

Reply:
3f98439bd3f624b6033ba065543e45e566aa2193